### ISSUE TITLE
.show-apps first/last-child pseudo-classes

### ISSUE THREAD

**Issue by zagortenay333** [micheleg/dash-to-dock#344]
Right now, `.show-apps` has both pseudo-classes regardless of whether it is displayed at the top/bottom or left/right of the dash.

For theming purposes, it would be nice if it only had one of these classes based on where it is.

**Comment by micheleg** [comment]
Hi, as I don't think I'm touching those classes in the extension, could you confirm if the same behaviour occurs also with the default dash, i.e. disabling the extension?

**Comment by zagortenay333** [comment]
Yes, the normal dash's `.show-apps` has both pseudo-classes too. You can check it with shell's built-in inspector.

**Comment by micheleg** [comment]
I see. That is because the show apps icon has its own container, is not at the same level of all other launchers for reasons of keeping it's visibility in case of lack of space.

**Comment by zagortenay333** [comment]
Any way to fix it programmatically?

Also on the same note, all the `.app-well-app` classes have both first/last-child pseudo-classes too. Kinda renders them useless in this state.

**Comment by micheleg** [comment]
for the `.app-well-app`, it's the parent container which should have the right first/last-child pseudo-class. It should be of class `.dash-item-container`.

Being pseudo classes, I guess they are "applied" by the Shell toolkit, and I'm not sure there is a way to force it. Moreover, I'd keep the css structure as close a possible to the defualt dash to avoid incompatibilities, What I could do instead is to add a specific class, maybe to the #dashtodockContainer like `.beginning-show-apps` or something similar, the same way I do for the other additional classes (`.bottom`, `.extended` _etc_ ).  This would allow easily to style the button in the two cases.

**Comment by zagortenay333** [comment]
Yeah, would be fine as well.

Btw, the show-apps button is inside a `.dash-item-container` and has the last-child pseudo-class while there is another `.dash-item-container` which also has one. So I can't use it properly the there either. :disappointed_relieved:

**Comment by zagortenay333** [comment]
Ah, nvm. It's possible to use the `dashtodockDashScrollview` container to namespace it.

**Comment by micheleg** [comment]
I would not rely on that too much. I can't guarantee I won't change the internal structure of the containers involved. I tried to document a bit the classes available here: http://micheleg.github.io/dash-to-dock/theming.html.
